platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_configurations.adoc

```diff
@@ -0,0 +1,145 @@
+== Understanding dependency configurations
+
+Gradle represents the scope of a dependency with the help of a link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[Configuration].
+Every configuration can be identified by a unique name.
+
+Many Gradle plugins add pre-defined configurations to your project.
+
+The Java Library plugin is used to define a project that produces a Java library.
+The plugin adds many configurations.
+These configurations represent the various classpaths needed for source code compilation, executing tests, and more:
+
+[cols="~,~,~"]
+|===
+|Configuration Name |Description |Used to:
+
+|`implementation`
+|Dependencies required for both compilation and runtime.
+|Declare Dependencies
+
+|`compileOnly`
+|Dependencies needed only for compilation, not included in runtime or publication.
+|Declare Dependencies
+
+|`runtimeOnly`
+|Dependencies needed only at runtime, not included in the compile classpath.
+|Declare Dependencies
+
+|`compileClasspath`
+|Combines `implementation` and `compileOnly` dependencies, used for compiling source code.
+|Resolve Dependencies
+
+|`runtimeClasspath`
+|Combines `implementation` and `runtimeOnly` dependencies, used for running the application.
+|Resolve Dependencies
+
+|`testImplementation`
+|Dependencies required for compiling and running tests.
+|Declare Dependencies
+
+|`testCompileOnly`
+|Dependencies needed only for test compilation.
+|Declare Dependencies
+
+|`testRuntimeOnly`
+|Dependencies needed only for running tests.
+|Declare Dependencies
+
+|`testCompileClasspath`
+|Combines `testImplementation` and `testCompileOnly` dependencies, used for compiling test sources.
+|Resolve Dependencies
+
+|`testRuntimeClasspath`
+|Combines `testImplementation` and `testRuntimeOnly` dependencies, used for running tests.
+|Resolve Dependencies
+
+|`apiElements`
+|Combines .
+|Consume Dependencies
+
+|`runtimeElements`
+|Combines .
+|Consume Dependencies
+
+|===
+
+=== Dependency declaration Configurations
+
+The *dependency declaration configurations* (`compileOnly`, `implementation`, `runtimeOnly`) focus on declaring and managing dependencies based on their usage (compile time, runtime, API exposure):
+
+[source,kotlin]
+----
+dependencies {
+    implementation("com.google.guava:guava:30.1.1-jre")     // Implementation dependency
+    compileOnly("org.projectlombok:lombok:1.18.20")         // Compile-only dependency
+    runtimeOnly("mysql:mysql-connector-java:8.0.23")        // Runtime-only dependency
+}
+----
+
+=== Other Configurations
+
+As you can see in the table above, there are other types of configurations, but they are not used to declare dependencies.
+
+The *resolving configurations* (`runtimeClasspath`, `compileClasspath`) focus on resolving dependencies needed at different stages of the build process.
+The *consuming configurations* (`apiElements`, `runtimeElements`) focus on exposing the artifacts and dependencies that a project produces for other projects to consume.
```

Comment:
Calling these "consuming" configurations make them seem like they are _performing_ consumption, when in fact they are intended to _be consumed_. 

For this reason, we generally them "consumable configurations".